Zend/zend_interfaces.stub.php

```diff
@@ -48,11 +48,11 @@ public function offsetUnset(mixed $offset);
 
 interface Serializable
 {
-    /** @return string */
+    /** @return string|null */
     public function serialize();
 
     /** @return void */
-    public function unserialize(string $data);
+    public function unserialize(?string $data);
```

Comment:
This one isn't right, `unserialize()` accepts on string.

Reply:
https://3v4l.org/JlkBD it accepts null as well, and serialize is documented with return type of `string|null`, thus unserialize should accept the same type

Reply:
You can, of course, make your unserialize() implementation accept whatever you want, but the interface only requires that it accepts strings. `unserialize()` will only ever call it with a string.

Reply:
You meant `unserialize` function (not the `Serializable::unserialize` method)... Then this change is right to match the return type of `Serializable::serialize` return type.

Reply:
No, this change is still wrong. If Serializable::serialize() returns null, the whole object is replaced with null and Serializable::unserialize() will never be called. Serializable::unserialize() will never be called with a null argument by PHP. If you manually call it with a null argument, that's your problem, it is not supported by the interface contract.